lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -2,8 +2,11 @@
   <name><%= @name %></name>
   <uuid><%= @uuid %></uuid>
   <memory><%= @memory_size %></memory>
-  <vcpu><%= @cpus %></vcpu>
-
+  <vcpu
+      <% if @cpuset %>
+          cpuset='<%= @cpuset %>'
+      <% end %>
```

Comment:
if you change the above to:
```
    <%- if @cpuset -%>
        cpuset='<%= @cpuset %>'
    <%- end -%>
```
it should correctly pass the template rendering test.

Reply:
There were still newlines after those modifications, so I changed it up until I produced the result expected by the tests. I also added some expected output to the unit test for all settings.